src/plugins/intel_gpu/src/plugin/transformations/disable_fp16_comp_for_periodic_funcs.cpp

```diff
@@ -60,6 +74,23 @@ ov::intel_gpu::DisableFP16CompressionForPeriodicFuncs::DisableFP16CompressionFor
 
     ov::matcher_pass_callback callback = [&](ov::pass::pattern::Matcher& m) {
         auto node = m.get_match_root();
+        // Only disable FP16 compression for FP32 data type,
+        // because sin/cos with fp64 should be converted to sin/cos with fp32.
+        if (node->get_output_element_type(0) != ov::element::f32) {
+            return false;
+        }
+
+        // If sin/cos propagates through a user node,
+        // it means that the sin/cos node is directly followed by a convert node from the user.
+        // Therefore, disable_fp16_compression is not set in this case.
```

Comment:
Could you elaborate more how the first sentence("if sin/cos propates through a user node") is connected to second sentence? ("sin/cos node is directly followed by a convert node")

Reply:
Modified the comments